x-pack/plugins/apm/common/data_view_constants.ts

```diff
@@ -5,5 +5,11 @@
  * 2.0.
  */
 
-// value of const needs to be backwards compatible
-export const APM_STATIC_DATA_VIEW_ID = 'apm_static_index_pattern_id';
+export const DO_NOT_USE_LEGACY_APM_STATIC_DATA_VIEW_ID =
+  'apm_static_index_pattern_id';
+
+const APM_STATIC_DATA_VIEW_ID_PREFIX = 'apm_static_data_view_id';
+
+export function getDataViewId(spaceId: string) {
+  return `${APM_STATIC_DATA_VIEW_ID_PREFIX}_${spaceId}`;
+}
```

Comment:
@shahzad31 What do you suggest we do for the UX app now that the data view will be space aware (one data view per space)?

https://github.com/sqren/kibana/blob/aff71b5b646c696f9f958c1dfc6f303a2309e5a0/x-pack/plugins/ux/common/index_pattern_constants.ts#L8

Reply:
Same applies to exploratory view

Reply:
@nreese Same applies to Maps. Changing the id for the APM data view will likely break APM in Maps. 

https://github.com/elastic/kibana/blob/5c44377de17f8a2da8790c5308822e1fc4d8aa95/x-pack/plugins/maps/public/classes/layers/wizards/solution_layers/observability/create_layer_descriptor.ts#L41

Reply:
from what i remember UX and Exp view are based on ad-hoc data views.

Reply:
Okay, so the code I linked to (and the static index pattern) is no longer used?

Reply:
Yeah that seems like redundant code.